### DISCUSSION TITLE
Import all icons from a single type

### DISCUSSION THREAD

**Discussion by juancarlosgzs** [ng-icons/ng-icons#64] [Q&A] [answered]
Hello! First of all: Great job! Loving it so far! This is something Angular has been missing for quite a lot of time and you have plenty of options in other frameworks. 

Anyway, I was wondering if there is a way to get all the icons from a single library such as Ionicons or any other, without having to import individually every single icon.

Thanks!

**Comment by ashley-hunter** [comment, marked as answer]
Hi! Thank you very much, I'm glad you are finding it useful!

Yes, you can indeed, you just need to import using `*` from the package:

```ts
import * as ionicons from '@ng-icons/ionicons';
```

You then register the icons like this:

```ts
provideIcons(ionicons)
```

While convenient, be warned that icons individually imported to improve bundle size, importing all tabler icons will have an effect on you applications size:

<img width="1146" alt="image" src="https://github.com/ng-icons/ng-icons/assets/20795331/942a0b6d-dffd-44f9-9b4b-5e6eab935eff">

A better way might be to have an single `provideIcons` at the root of your application when add all icons you are actually using there, that way you only have to import them once rather than in every component/ng module. This would help with reducing the bundle size.

Regardless, here is an example of how to import all icons.

https://stackblitz.com/edit/stackblitz-starters-vesc9p

**Reply by juancarlosgzs** [reply]
Thanks a lot for such a complete and fast response!

**Reply by ashley-hunter** [reply]
No problem 🙂
